### PR TITLE
Add summary feature tests

### LINKED ISSUE #302: Add tests for summary feature

### PR DESCRIPTION
**Pull request by ngzhenteng**
Fix date formatting in MC and Consultation cards
Improve date value error message to be more verbose
Fixes #302

### CODE REVIEW

File: src/test/java/seedu/jarvis/model/summary/SummaryTest.java [hunk 1]
```diff
@@ -1,12 +1,137 @@
...
+
+    @Test
+    public void generateTasksString_doesNotEndWithComma() {
+        String generatedString = summaryWithSpecificValues.generateTaskString();
+        assertFalse(generatedString.endsWith(", "));
+    }
+
+    // Integration tests
+    @Test
+    public void constructor_createsHashMapWithDefaultValues() {
```
Comment (wilinetan) [inline src/test/java/seedu/jarvis/model/summary/SummaryTest.java:123]: Thanks for adding a mixture of unit and integration tests

File: src/main/java/seedu/jarvis/model/consultation/Consultation.java [hunk 2]
```diff
@@ -28,6 +30,10 @@ public LocalDateTime getDateAndTime() {
         return dateAndTime;
     }
 
+    public String getFormattedDateTime() {
+        return dateAndTime.format(DateTimeFormatter.ofLocalizedDateTime(FormatStyle.SHORT));
+    }
+
```
Comment (wilinetan) [inline src/main/java/seedu/jarvis/model/consultation/Consultation.java:36]: Good method provided to abstract the details!

**Review by wilinetan — COMMENTED** [review]
thanks for adding tests and fixing the dates

File: src/main/java/seedu/jarvis/logic/commands/add/AddCommand.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
     public static final String MESSAGE_WRONG_DATETIME_FORMAT = "The date time format is incorrect. Correct format: "
             + "d/YYYY-MM-DD t/HH:MM";
     public static final String MESSAGE_INVALID_DATETIME = "The provided date and time is incorrect.\nBe careful of "
-            + "leap years and the number of days in a specific month";
+            + "leap years, number of months in a year and the number of days in a specific month.";
```
Comment (Asuraxsoul) [inline src/main/java/seedu/jarvis/logic/commands/add/AddCommand.java:30]: Thanks for fine tuning the error messages.

File: src/main/java/seedu/jarvis/model/consultation/Consultation.java (hunk 2 above)
Comment (Asuraxsoul) [reply src/main/java/seedu/jarvis/model/consultation/Consultation.java:36]: thanks for making the date and time consistent

**Review by Asuraxsoul — COMMENTED** [review]
LGTM

File: src/main/java/seedu/jarvis/model/consultation/Consultation.java (hunk 2 above)
Comment (wireseo) [reply src/main/java/seedu/jarvis/model/consultation/Consultation.java:36]: Agreed, thanks for the effort!

**Pull request merged**